proposals/0296-increase-tx-size.md

```diff
@@ -0,0 +1,241 @@
+developers to create new applications and use cases that were previously
+unfeasible.
+
+## New Terminology
+
+- 'v1 transaction' - A new transaction format that is designed to enable larger
+  transactions sizes while not having the address lookup table features 
+  introduced in v0 transactions.
+
+## Detailed Design
+
+Increasing the transaction size limit above the current MTU max size is
+possible with the use of QUIC. QUIC's RFC 9000 specification does not have an
+explicit maximum stream size, allowing for larger transactions to be sent.
+
+A new transaction format, `v1`, is proposed to enable larger transaction sizes.
+The `v1` transaction format would be:
+
+```
+VersionByte (u8) - =129 to distinguish from legacy/v0 formats
+LegacyHeader (u8, u8, u8) -- Required signatures from the current
+`MessageHeader` type
+NumInstructions (u8)
+TransactionConfigMask (u32) -- Bitmask of which config requests are present.
+LifetimeSpecifier [u8; 32]
+NumAddresses (u8)
+Addresses [[u8; 32]] -- Length matches NumAddresses
+ConfigRequests [u8] -- Array of request values. (section size is popcount 
+  TransactionConfigMask bytes). See section TransactionConfigMask for details.
+InstructionHeaders [(u8, u8, u16)] -- Length of NumInstructions. Values are 
+  (ProgramAccountIndex, NumInstructionAccounts, NumInstructionDataBytes)
+InstructionPayloads [InstructionPayload] -- Length = NumInstructions.
+  Each InstructionPayload is the concatenation of the following byte arrays:
+    InstructionAccountIndexes [u8] -- Length = NumInstructionAccounts from the 
+    corresponding InstructionHeader
+    InstructionData [u8] -- Length = NumInstructionDataBytes from the 
+    corresponding InstructionHeader
+```
+
+The [TRAILING DATA SECTION] contains instruction's serialized account indexes
+and instruction data bytes,
+in the order they appear in the transaction. Each instruction's account indexes
+must appear before its data bytes.
+For example, if there are two instructions:
+
+1. `Instruction { account_indexes: [0, 1], data: [2, 3, 4] }`
+2. `Instruction { account_indexes: [5, 6], data: [7, 8] }`
+
+The TRAILING DATA SECTION must look like this:
+
+```
+[0, 1, 2, 3, 4, 5, 6, 7, 8]
+```
+
+Lengths are serialized in the earlier `Ixs` section, and so are not repeated in
+the trailing data section.
+
+This new `v1` transaction format notably does not include address lookup
+tables.
+
+### Transaction Constraints
+
+Any transaction violating these constraints will be considered invalid and will 
+not be included in the chain. Violations are considered sanitization failures:
+
+| max | value | new/existing |
+| --- | --- | --- |
+| max transaction size | 4096 | new |
+| max num signatures per transaction | 42 | new |
+| max num accounts | 96 | new |
+| max num instructions | 64 | old |
+| max accounts/instruction | 255 | new |
+
+While the above lists sanitization failures, we will keep the current runtime 
+max limit of 12 signatures per transaction.
```

Comment:
We actually don't have an explicit runtime max limit of 12 signatures. It's implicit right now. We should say that we are _adding_ a runtime max limit of 12 signatures to match the previous implicit limit from the previous max tx size.

We should also say that the runtime limit for the max number of tx accounts is unchanged from 64.

Reply:
Updated to account for this being an already implicit limit, as well as noting the current runtime limit for the max number of accounts